### PR TITLE
Allow flushing of item by key before reading

### PR DESCRIPTION
**Pull request by SpacePossum**
https://github.com/GeckoPackages/GeckoSilexConfigService/pull/6

### CODE REVIEW

**Comment by SpacePossum** [comment]
ping @tiemevanveen 
This should replace https://github.com/GeckoPackages/GeckoSilexConfigService/pull/6 and allow you to flush a configuration item by key before calling the configuration itself by said key.

**Comment by tiemevanveen** [comment]
Haha, I was like, huh, did I missed this.

Will check after work tonight!

File: src/Config/ConfigLoader.php [hunk 1]
```diff
@@ -90,10 +91,11 @@ public function __construct(Application $app, $dir = null, $format = '%key%.json
      * @return array
      *
      * @throws FileNotFoundException
+     * @throws IOException
      */
     public function get($key)
     {
-        if (array_key_exists($key, $this->config)) {
+        if (isset($this->config[$key])) {
```
Comment (tiemevanveen) [inline src/Config/ConfigLoader.php]: Haha why this change?

**Comment by tiemevanveen** [comment]
Nice work, looks good and works like a charm!

I could not help myself and separated the logic in 2 commits, one of the changes I proposed and one of your changed. https://github.com/GeckoPackages/GeckoSilexConfigService/pull/10 feel free to merge either #9 or #10 :)

File: src/Config/ConfigLoader.php (hunk 1 above)
Comment (SpacePossum) [inline src/Config/ConfigLoader.php]: micro optimize ;)
`isset` is a language construct and `array_key_exists` a function, while functional not 100% the same in this case it will work and is a bit faster ^-^

**Comment by SpacePossum** [comment]
@ see https://github.com/GeckoPackages/GeckoSilexConfigService/pull/10
